tests/test_external_roundtrip.rs

```diff
@@ -0,0 +1,116 @@
+extern crate env_logger;
+#[macro_use]
+extern crate log;
+extern crate tempdir;
+
+use tempdir::TempDir;
+
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+/// A macro used to simplify debugging.
+const DEBUG: bool = true;
+
+fn get_path(root: &Path, components: &[&str]) -> PathBuf {
+    let mut root = root.to_path_buf();
+    for component in components {
+        root.push(component)
+    }
+    root
+}
+
+fn run(mut command: std::process::Command) -> Result<(), std::io::Error> {
+    let output = command.output()?;
+    if DEBUG || !output.status.success() {
+        debug!(target: "test", "Command: {command:?}\n** out:{out}\n** err:{err}",
+            command = command,
+            out =  String::from_utf8(output.stdout).unwrap_or_else(|_| "<cannot decode output>".to_string()),
+            err =  String::from_utf8(output.stderr).unwrap_or_else(|_| "<cannot decode output>".to_string())
+        );
+    }
+    assert!(output.status.success());
+    Ok(())
+}
+
+fn perform_test(root: &Path, prefix: &str, source: &str) {
+    let path_in_source = std::path::Path::new(source).to_path_buf();
```

Comment:
I don't think you need `to_path_buf` here since `fs::copy` accepts just `&Path` too.